Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -286,7 +289,32 @@ std::optional<InteractionRegion> interactionRegionForRenderedRegion(RenderObject
     OptionSet<InteractionRegion::CornerMask> maskedCorners { };
     std::optional<Path> clipPath = std::nullopt;
 
-    if (const auto& renderBox = dynamicDowncast<RenderBox>(regionRenderer)) {
+    if (const auto& renderShape = dynamicDowncast<LegacyRenderSVGShape>(regionRenderer)) {
+        auto& svgElement = renderShape->graphicsElement();
+        auto path = svgElement.toClipPath();
+
+        auto shapeBoundingBox = svgElement.getBBox(SVGLocatable::DisallowStyleUpdate);
+        path.translate(FloatSize(-shapeBoundingBox.x(), -shapeBoundingBox.y()));
+
+        auto* svgSVGElement = svgElement.ownerSVGElement();
+        if (svgSVGElement) {
+            FloatSize size = svgSVGElement->currentViewportSizeExcludingZoom();
+            auto viewBoxTransform = svgSVGElement->viewBoxToViewTransform(size.width(), size.height());
+            shapeBoundingBox = viewBoxTransform.mapRect(shapeBoundingBox);
+            path.transform(AffineTransform::makeScale(FloatSize(viewBoxTransform.xScale(), viewBoxTransform.yScale())));
+
+            // Position to respect clipping. `rect` is already clipped but the Path is complete.
+            auto clipDeltaX = rect.width() - shapeBoundingBox.width();
+            auto clipDeltaY = rect.height() - shapeBoundingBox.height();
+            if (shapeBoundingBox.x() >= 0)
+                clipDeltaX = 0;
+            if (shapeBoundingBox.y() >= 0)
+                clipDeltaY = 0;
+            path.translate(FloatSize(clipDeltaX, clipDeltaY));
+        }
+
```

Comment:
Couldn't find a more elegant solution to get the transform we need here, but opened to suggestions!